### ISSUE TITLE
Math structure

### ISSUE THREAD

**Issue by julianhyde** [julianhyde/morel#88]
Implement the "math" structure of the standard basis library. Per [SML basis](https://smlfamily.github.io/Basis/math.html) it has the following interface:
```
type real
val pi : real
val e : real
val sqrt : real -> real
val sin : real -> real
val cos : real -> real
val tan : real -> real
val asin : real -> real
val acos : real -> real
val atan : real -> real
val atan2 : real * real -> real
val exp : real -> real
val pow : real * real -> real
val ln    : real -> real
val log10 : real -> real
val sinh : real -> real
val cosh : real -> real
val tanh : real -> real
```

Example use:
```
Math.ln 1.0;
val it = 0.0 : real;
```

**Comment by julianhyde** [comment]
Work in progress: https://github.com/julianhyde/morel/tree/0088-math

**Comment by julianhyde** [comment]
Fixed in ff4277505514af4cc8163e0be4c3a834fb8f7807.

In the previous commit a2e20dd75553438ebbf2cc12499939cd1414deef, we started on the `Real` structure (supporting floating point arithmetic), adding constants `Real.posInf` and `Real.negInf`. More of the `Real` structure will follow shortly.
